src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -16,17 +16,32 @@
 
 namespace hyrise {
 
-void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+    void print_execution_time(const std::chrono::time_point<std::chrono::high_resolution_clock> start){
+        auto stop = std::chrono::high_resolution_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::microseconds>(stop - start);
+        std::cout << "Thread execution time: " << duration.count() << " microseconds." << std::endl;
+    }
+
+void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                          std::atomic<bool>& verbose)  {
+    auto start = std::chrono::high_resolution_clock::now();
   const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
   const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
-  lseek(fd, from * uint32_t_size, SEEK_SET);
-  Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
-         close_file_and_return_error_message(fd, "Read error: ", errno));
+
+
+    lseek(fd, from * uint32_t_size, SEEK_SET);
+    Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
+           close_file_and_return_error_message(fd, "Read error: ", errno));
+
+  if (verbose){
+      print_execution_time(start);
+  }
```

Comment:
Can't we do these things in `Setup()` and `Teardown()`? I feel like that is a more elegant approach, except there is a reason not do it. Directly I have in my head, that Google Benchmark differs in its execution times of the standard "`IO Methods`" and the `Setup()` and `Teardown()`.

Reply:
I think this is not possible because we want to measure the execution time of each thread created during our benchmark.

Reply:
Agree with Jan on this one.